utils/http.go

```diff
@@ -55,6 +55,18 @@ func (root *rootHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	ctx = context.WithValue(ctx, "http.request", r)
 
+	if root.auth != nil {
+		var err error
+		access := buildAccessRecords(vars["imageName"], root.actions...)
+		if ctx, err = root.auth.Authorized(ctx, access...); err != nil {
+			if err, ok := err.(auth.Challenge); ok {
+				err.ServeHTTP(w, r)
```

Comment:
The auth challenge sets the `WWW-Authenticate` header, but it does not set the status code. Currently it returns a 200 with the header.